mcp2210-core.c

```diff
@@ -709,6 +712,358 @@ static inline int eeprom_read_complete(struct mcp2210_cmd *cmd_head,
 }
 #endif /* CONFIG_MCP2210_CREEK */
 
+#ifdef CONFIG_OF
```

Comment:
It would appear that using `CONFIG_OF` will create a problem for people building the kernel with OF enabled, but not using it in this driver. This should be `CONFIG_MCP2210_OF` and depend upon `CONFIG_OF` in `Kconfig`. Of course, `Kconfig` isn't actually being used yet, but if nothing else it documents config parameters. As such, we should probably have built-time sanity checks for them in some function that's always compiled, like probe:

```
BUILD_BUG_ON(IS_ENABLED(CONFIG_MCP2210_OF) && !IS_ENABLED(CONFIG_OF));
BUILD_BUG_ON(IS_ENABLED(CONFIG_MCP2210_OF) && IS_ENABLED(CONFIG_MCP2210_CREEK));
```